tests/internal/utils.c

```diff
@@ -110,8 +110,42 @@ void test_url_split()
     }
 }
 
+void test_write_str()
+{
+    char buf[10];
+    int size = sizeof(buf);
+    int off;
+    int ret;
+
+    off = 0;
+    ret = flb_utils_write_str(buf, &off, size, "a", 1);
+    TEST_CHECK(ret == FLB_TRUE);
+    TEST_CHECK(memcmp(buf, "a", off) == 0);
+
+    off = 0;
+    ret = flb_utils_write_str(buf, &off, size, "\n", 1);
+    TEST_CHECK(ret == FLB_TRUE);
+    TEST_CHECK(memcmp(buf, "\\n", off) == 0);
+
+    off = 0;
+    ret = flb_utils_write_str(buf, &off, size, "\xe3\x81\x82", 3);
+    TEST_CHECK(ret == FLB_TRUE);
+    TEST_CHECK(memcmp(buf, "\\u3042", off) == 0);
+
+    // Error: buffer too small
+    off = 0;
+    ret = flb_utils_write_str(buf, &off, size, "aaaaaaaaaaa", 11);
+    TEST_CHECK(ret == FLB_FALSE);
+
+    // Error: invalid UTF-8
+    off = 0;
+    ret = flb_utils_write_str(buf, &off, size, "\xe3", 1);
+    TEST_CHECK(off == 0);
+}
```

Comment:
What should be the result of "\xe3\x81\x82\xe3"?

"\\u3042"? or nothing?

It would be good for the test coverage to clarify the intention.

Reply:
Thank you. I updated the test case clarifying the intention.